src/compiler/checker.ts

```diff
@@ -7301,11 +7318,8 @@ namespace ts {
                 if (type.flags & TypeFlags.Union) {
                     return getUnionType(map((<UnionType>type).types, getWidenedConstituentType));
                 }
-                if (isArrayType(type)) {
-                    return createArrayType(getWidenedType((<TypeReference>type).typeArguments[0]));
-                }
-                if (isTupleType(type)) {
-                    return createTupleType(map(type.elementTypes, getWidenedType));
+                if (isArrayType(type) || isTupleType(type)) {
+                    return createTypeReference((<TypeReference>type).target, map((<TypeReference>type).typeArguments, getWidenedType));

```

Comment:
half the uses of `isArrayType` require the type assertion, so I think it's worth it.

---
In reply to: [75712577](https://github.com/Microsoft/TypeScript/pull/10466#discussion_r75712577) [](ancestors = 75712577)